backend/tests/v5/e2e/routes/teamspaces/projects/models/common/groups.test.js

```diff
@@ -0,0 +1,215 @@
+			[`the ${modelLabel} does not exist`, createRoute({ modelId: 'dslfkjds' }), false, modelNotFoundRes],
+			[`the model is not a ${modelLabel}`, createRoute({ modelId: incorrectModelTypeId }), false, modelNotFoundRes],
+			[`the user does not have permissions to access the ${modelLabel}`, createRoute({ apiKey: users.noProjectAccess.apiKey }), false, templates.notAuthorized],
+			['the data is not of the right schema', createRoute(), false, templates.invalidArguments, { groups: 1 }],
+			['the groups array is empty', createRoute(), false, templates.invalidArguments, { groups: [] }],
+			['the groups requested exists (1)', createRoute(), true, { groups }],
+			['the groups requested exists (2)', createRoute(), true, { groups: [groups[0]] }, { groups: [groups[0]._id] }],
+			['the groups requested doesn\'t exist', createRoute({ modelId: modelWithNoGroups }), true, { groups: [] }],
+		];
+	};
+
+	const runTest = (desc, route, success, expectedRes, post = postData) => {
+		test(`Should ${success ? 'succeed' : `fail with ${expectedRes.code}`} if ${desc}`, async () => {
+			const expectedStatus = success ? templates.ok.status : expectedRes.status;
+			const res = await agent.post(route)
+				.send(post)
+				.expect(expectedStatus);
+			if (success) {
+				expect(res.body).toEqual(expectedRes);
+			} else {
+				expect(res.body.code).toEqual(expectedRes.code);
+			}
+		});
+	};
+
+	describe('Export groups', () => {
+		beforeAll(async () => {
+			const models = [container, fed, containerNoGroups, fedNoGroups];
+			await ServiceHelper.db.createTeamspace(teamspace, [users.tsAdmin.user]);
+
+			const userProms = Object.keys(users).map((key) => ServiceHelper.db.createUser(users[key], key === 'nobody' ? [] : [teamspace]));
+			const modelProms = models.map((model) => ServiceHelper.db.createModel(
+				teamspace,
+				model._id,
+				model.name,
+				model.properties,
+			));
+			return Promise.all([
+				...userProms,
+				...modelProms,
+				ServiceHelper.db.createProject(teamspace, project.id, project.name, models.map(({ _id }) => _id)),
+				ServiceHelper.db.createGroups(teamspace, container._id, groups),
+				ServiceHelper.db.createGroups(teamspace, fed._id, groups),
+			]);
+		});
+		describe.each(generateTestData(true))('Federations', runTest);
+		describe.each(generateTestData())('Containers', runTest);
+	});
+};
+
+const testImportGroups = () => {
+	const users = {
+		tsAdmin: ServiceHelper.generateUserCredentials(),
+		noProjectAccess: ServiceHelper.generateUserCredentials(),
+		nobody: ServiceHelper.generateUserCredentials(),
+	};
+
+	const teamspace = ServiceHelper.generateRandomString();
+
+	const project = {
+		id: ServiceHelper.generateUUIDString(),
+		name: ServiceHelper.generateRandomString(),
+	};
+
+	const container = ServiceHelper.generateRandomModel();
+	const fed = ServiceHelper.generateRandomModel({ isFederation: true });
+
+	const groups = [
+		ServiceHelper.generateGroup(teamspace, container._id),
```

Comment:
shouldn't we include some federation groups as well ?

Reply:
that model id is for the objects identifier (i.e. given a geometry UUID, which model did it come from) - this function just generates some group data, it doesn't save it to a database

federation is just a federated list of models, it'll never own any geometry (i.e. it'll always come from a container)